### DISCUSSION TITLE
Questions about the integration of torque vectoring and MPC

### DISCUSSION THREAD

**Discussion by chentairan** [alexliniger/MPCC#75] [Q&A]
Hi @alexliniger ,

I am recently reading the paper "A Holistic Motion Planning and Control Solution to Challenge a Professional Racecar Driver". I have some questions about the relationship between yaw moment $\tau_{z}$ and Mtv (from MPC-Curv) in torque vectoring.

In the low level control section, the yaw moment is calculated by a simple p-controller $\tau_{z} = KP(\bar{r} - r)$, and I would like to know how this yaw moment is specifically used in torque vectoring.
Besides, It introduces `Mtv` as an input in MPC-Curv, but I don't see how this `Mtv` is used in the torque vectoring.

Best,
Tairan Chen

**Comment by alexliniger** [comment]
Sorry this answer is very late, with the original code it would be very complex to change the model.
However, with the new automatic differentiation branch it should be straightforward to change the model of the car, I already added brakes, but also adding 4-wheel drive or torque vectoring should be easy.

I will add the installation instructions soon.
